### PR TITLE
add testing for git worktree

### PR DESCRIPTION
**Pull request by yiweny**
add a test case to show that removing jgit allows git worktree.

### CODE REVIEW

File: src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy [hunk 1]
```diff
@@ -81,7 +81,39 @@ class GitVersionPluginTests extends Specification {
...
+        File initialGitIgnoreFile = new File(initialDir, ".gitignore")
+        initialGitIgnoreFile.createNewFile()
+        initialGitIgnoreFile << '.gradle\n'
+        Git git = new Git(initialDir, true)
+        git.runGitCommand("init", initialDir.toString())
+        git.runGitCommand("add", ".")
+        git.runGitCommand("commit","-m", "'initial commit'")
+        git.runGitCommand("tag", "-a", "1.0.0", "-m", "1.0.0")
+        git.runGitCommand("branch", "hotfix")
+        git.runGitCommand("worktree", "add", "../hotfix", "hotfix")
```
Comment (pkoenig10) [inline src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy]: I think `git worktree add` only takes a single path argument. I'm not sure I understand what this command is supposed to be doing.

I'd also prefer if we avoided names with specific connotations like "hotfix". There's nothing specific to "hotfixes" about worktree support.

Comment (yiweny) [reply src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy]: The command creates a worktree at `../hotfix` and checks out the `hotfix` branch. 
I fixed my code to use a more neutral name.

File: src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy [hunk 2]
```diff
@@ -81,7 +81,39 @@ class GitVersionPluginTests extends Specification {
...
+
+        then:
+        buildResult.output.contains(":printVersion\n1.0.0\n")
+    }
+
+    def 'git describe works when using worktree' () {
+        given:
+        File rootFolder = temporaryFolder
+        projectDir = Files.createDirectories(rootFolder.toPath().resolve('level1/newbranch')).toFile()
+        File initialDir = Files.createDirectories(rootFolder.toPath().resolve('level1/level2')).toFile()
```
Comment (pkoenig10) [inline src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy]: nit: Clean up this levels stuff, that's not what we're testing here

```suggestion
        projectDir = Files.createDirectories(rootFolder.toPath().resolve('original')).toFile()
        File initialDir = Files.createDirectories(rootFolder.toPath().resolve('worktree')).toFile()
```

Comment (yiweny) [reply src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy]: fixed based on the suggestion.

**Review by pkoenig10 — APPROVED** [review]

**Pull request merged**